client/js/components/assessments/item.jsx

```diff
@@ -7,211 +7,124 @@ import UniversalInput         from "./universal_input";
-    questionCount    : React.PropTypes.number.isRequired,
-    messageIndex     : React.PropTypes.number.isRequired,
-    confidenceLevels : React.PropTypes.bool.isRequired,
-    outcomes         : React.PropTypes.object
+    question                   : React.PropTypes.object.isRequired,
+    currentItemIndex           : React.PropTypes.number.isRequired,
+    questionCount              : React.PropTypes.number.isRequired,
+    messageIndex               : React.PropTypes.number.isRequired,
+    confidenceLevels           : React.PropTypes.bool.isRequired,
+    nextQuestion               : React.PropTypes.func.isRequired,
+    prevQuestion               : React.PropTypes.func.isRequired,
+    confidenceLevel            : React.PropTypes.func.isRequired,
+    submitAssessment           : React.PropTypes.func.isRequired,
+    outcomes                   : React.PropTypes.object,
   };
 
   nextButtonClicked(e){
     e.preventDefault();
-    this.setState({unAnsweredQuestions: null});
-    AssessmentActions.nextQuestion();
-    this.setState({showMessage: false});
+    this.props.nextQuestion();
   }
 
   previousButtonClicked(e){
     e.preventDefault();
-    this.setState({unAnsweredQuestions: null});
-    AssessmentActions.previousQuestion();
-    this.setState({showMessage: false});
-  }
-
-  confidenceLevelClicked(e, currentItemIndex){
-    // e.preventDefault();
-    //
-    // if(this.props.selectedAnswerId && this.props.selectedAnswerId.length > 0){
-    //   AssessmentActions.selectConfidenceLevel(e.target.value, currentItemIndex);
-    //   if(this.props.currentItemIndex == this.props.questionCount - 1 && this.props.settings.assessmentKind.toUpperCase() == "FORMATIVE"){
-    //     this.submitButtonClicked();
-    //   } else {
-    //     AssessmentActions.nextQuestion();
-    //     this.setState({showMessage: false});
-    //   }
-    // } else {
-    //   this.setState({showMessage: true});
-    // }
-    // if(document.getElementById("focus")){document.getElementById("focus").focus();}
+    this.props.prevQuestion();
   }
 
   submitButtonClicked(e){
-    e && e.preventDefault();
-    AssessmentActions.selectQuestion(this.props.currentItemIndex);
-    var complete = this.checkCompletion();
-    if(complete === true){
-      window.onbeforeunload = null;
-      AssessmentActions.submitAssessment(this.props.assessment.id, this.props.assessment.assessmentId, this.props.allQuestions, this.props.allStudentAnswers, this.props.settings, this.props.outcomes);
-    }
-    else {
-      this.setState({unAnsweredQuestions: complete});
-    }
-  }
-
-  checkCompletion(){
-    var questionsNotAnswered = [];
-    var answers = this.props.allStudentAnswers;
-    for (var i = 0; i < answers.length; i++) {
-      if(answers[i] == null || answers[i].length == 0){
-
-        questionsNotAnswered.push(i+1);
-      }
-    };
-    if(questionsNotAnswered.length > 0){
-      return questionsNotAnswered;
-    }
-    return true;
-  }
-
-  getFooterNav(){
```

Comment:
This functionality was nearly identical getNavigationButtons, so I kept only one of the two.